### PR TITLE
Interfaces and Transactors along with their tests

### PR DESCRIPTION
**Pull request by srivatsan-ramesh**
@tenthousandfailures, @cfelton, @jck, @josyb, @dargor0

The interfaces were created referencing figure 1 of this [pdf](http://www.xilinx.com/support/documentation/application_notes/xapp495_S6TMDS_Video_Interface.pdf)
- AuxInterface - with a transactor which writes a given data onto the signals.
- VideoInterface - with a transactor that writes a given pixel.
- HDMIInterface - contains TMDS differential signals.

### CODE REVIEW

File: hdmi/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+__version__ = '1.0'
```
Comment (cfelton) [inline hdmi/__init__.py]: [Versions](http://effbot.org/pyfaq/how-does-the-python-version-numbering-scheme-work.htm) are always interesting and fun.  Review the link for a brief description of the version scheme used in Python.  During development, **pre** the first release, I like to increment the "bug" number and target the first release to be `0.1.0`.  At this point I would version it as `0.0.1`.

File: hdmi/interfaces/hdmi_interface.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+
+class HDMIInterface:
+
+    def __init__(self, clock):
+
+        """
+         This interface is the internal interface modeled after
+         the xapp495 external HDMI interface
+        """
```
Comment (cfelton) [inline hdmi/interfaces/hdmi_interface.py:18]: I would add more information here on the HDMI link.  Describe each signal and describe what is required to send a frame or video stream.  The same `doc-string` seems like it was cut-n-paste from the others, this is not the internal interfaces - it is the external, correct?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        yield prefix
+        prefix += "."
+        for _, name, ispkg in walk_packages(path,
+                                            prefix,
+                                            onerror=lambda x: x):
+            if ispkg:
+                yield name
+
+
+def find_packages():
```
Comment (cfelton) [inline setup.py]: Setuptools has a `find_packages` function, I have only seen a custom find packages used if `distutils` and `setuptools` support is desired.  Use the setuptools `find_packages` and/or use distutil if setuptools is not available.

**Comment by srivatsan-ramesh** [comment]
Shall I merge the pull request? @cfelton

**Comment by cfelton** [comment]
Yes, go for it.

**Pull request merged**